Algebroids/gap/PathCategories.gi

```diff
@@ -0,0 +1,1039 @@
+        
+        return IsEqualForObjects( C, Source( mor_1 ), Source( mor_2 ) )
+                  and IsEqualForObjects( C, Target( mor_1 ), Target( mor_2 ) )
+                    and MorphismDatum( mor_1 ) = MorphismDatum( mor_2 );
+        
+    end );
+    
+    ##
+    AddIsCongruentForMorphisms( C,
+      
+      function ( C, mor_1, mor_2 )
+        
+        return IsEqualForObjects( C, Source( mor_1 ), Source( mor_2 ) )
+                  and IsEqualForObjects( C, Target( mor_1 ), Target( mor_2 ) )
+                    and MorphismDatum( mor_1 ) = MorphismDatum( mor_2 );
+        
+    end );
+    
+    AddIdentityMorphism( C,
+      
+      function ( C, obj )
+        
+        return MorphismConstructor( C, obj, Pair( 0, [ ] ), obj );
+        
+    end );
+    
+    AddPreCompose( C,
+      
+      function ( C, mor_1, mor_2 )
+        
+        return MorphismConstructor( C,
+                      Source( mor_1 ),
+                      Pair( MorphismLength( mor_1 ) + MorphismLength( mor_2 ), Concatenation( MorphismSupport( mor_1 ), MorphismSupport( mor_2 ) ) ),
+                      Target( mor_2 ) );
+        
+    end );
+    
+    ## Random Methods, maybe they can be optimized!
+    ##
+    AddRandomObjectByInteger( C,
+      
+      function ( C, n )
+        
+        return Random( SetOfObjects( C ) );
+        
+    end );
+    
+    ##
+    AddRandomMorphismWithFixedSourceByInteger( C,
+      
+      function ( C, obj, n )
+        local s, t, m;
+        
+        s := ObjectIndex( obj );
+        t := Random( [ 1 .. NumberOfObjects( CapQuiver( C ) ) ] );
+        
+        m := ExternalHomsWithGivenLength( C, 0, n )[s][t];
+        
+        if m = [ ] then
+              return IdentityMorphism( C, obj );
+        else
+              return Random( m );
+        fi;
+        
+    end );
+    
+    ##
+    AddRandomMorphismWithFixedRangeByInteger( C,
+      
+      function ( C, obj, n )
+        local s, t, m;
+        
+        s := Random( [ 1 .. NumberOfObjects( CapQuiver( C ) ) ] );
+        t := ObjectIndex( obj );
+        
+        m := ExternalHomsWithGivenLength( C, 0, n )[s][t];
+        
+        if m = [ ] then
+              return IdentityMorphism( C, obj );
+        else
+              return Random( m );
+        fi;
+        
+    end );
+    
+    ##
+    AddRandomMorphismWithFixedSourceAndRangeByInteger( C,
+      
+      function ( C, obj_1, obj_2, n )
+        local s, t, p;
+        
+        s := ObjectIndex( obj_1 );
+        t := ObjectIndex( obj_2 );
+        
+        p := PositionProperty( [ 1 .. n + 1 ], i -> ExternalHomsWithGivenLength( C, n + 1 - i )[s][t] <> [ ] );
+        
+        if p = fail then
+              return fail;
+        else
+              return Random( ExternalHomsWithGivenLength( C, n + 1 - p )[s][t] );
+        fi;
+        
+    end );
+    
+    # Homomorphism Structure - Only for path categories with underlying acyclic quivers
```

Comment:
Please add an example that covers this case as well. It would be nice to have an example where the HomStructure is computable before taking the quotient and where one can see the difference in HomStructure after taking the quotient.